contrib/endpoints/src/api_manager/check_security_rules.cc

```diff
@@ -14,92 +14,350 @@
+const char kInvalidResponse[] = "Invalid JSON response from Firebase Service";
+
 // An AuthzChecker object is created for every incoming request. It does
 // authorizaiton by calling Firebase Rules service.
 class AuthzChecker : public std::enable_shared_from_this<AuthzChecker> {
  public:
-  AuthzChecker(std::shared_ptr<context::RequestContext> context,
-               std::function<void(Status status)> continuation);
+  // Constructor
+  AuthzChecker(ApiManagerEnvInterface *env,
+               auth::ServiceAccountToken *sa_token);
 
-  void Check();
+  // Check for Authorization success or failure
+  void Check(std::shared_ptr<context::RequestContext> context,
+             std::function<void(Status status)> continuation);
 
  private:
-  // Helper function to send a http GET request.
-  void HttpFetch(const std::string &url, const std::string &request_body,
+  // Helper method that invokes the test firebase service api.
+  void FirebaseCheck(std::string &ruleset_id,
+                     std::shared_ptr<context::RequestContext> context,
+                     std::function<void(Status status)> continuation);
+
+  // Parse the respose for GET RELEASE API call
+  Status ParseReleaseResponse(const std::string &json_str,
+                              std::string *ruleset_id);
+
+  // Parses the response for the TEST API call
+  Status ParseTestResponse(std::shared_ptr<context::RequestContext> context,
+                           const std::string &json_str);
+
+  // Builds the request body for the TESP API call.
+  Status BuildTestRequestBody(std::shared_ptr<context::RequestContext> context,
+                              std::string &result_string);
+
+  // Get the release name to used in Firebase API call
+  static const std::string GetReleaseName(
+      std::shared_ptr<context::RequestContext> request_context);
+
+  // Get the URL for the Release request.
+  static const std::string GetReleaseUrl(
+      std::shared_ptr<context::RequestContext> request_context);
+
+  static const std::string GetRulesetTestUri(std::string &ruleset_id);
+
+  // Invoke the HTTP call
+  void HttpFetch(const std::string &url, const std::string &method,
+                 const std::string &request_body,
                  std::function<void(Status, std::string &&)> continuation);
 
-  // Get Auth token for accessing Firebase Rules service.
-  const std::string &GetAuthToken();
+  // Get the auth token for Firebase service
+  const std::string &GetAuthToken() {
+    return sa_token_->GetAuthToken(
+        auth::ServiceAccountToken::JWT_TOKEN_FOR_FIREBASE);
+  }
 
-  // Request context.
-  std::shared_ptr<context::RequestContext> context_;
+  // Get Firebase specific operation Id based on the http Method.
+  static std::string GetOperation(std::string &httpMethod);
 
-  // Pointer to access ESP running environment.
-  ApiManagerEnvInterface *env_;
+  void SetProtoValue(::google::protobuf::Value &head, std::string key,
```

Comment:
this can be static.

I think all these 4 static functions can be out of the class.